articles/flask-database-integration-with-sqlalchemy/index.md

```diff
@@ -0,0 +1,329 @@
+```bash
+$ pip install -r requirement.txt
+``` 
+We also need to set up the Flask environment to develop from the default state which is the production environment.
+
+```bash
+$ export FLASK_ENV=”DEVELOPMENT”
+```
+
+The above terminal command sets the Flask environment as a development environment. Now, we can run our Flask app with the following command.
+
+```bash
+$ flask run
+```
+
+Our flask web application should be live, follow the host URL(127.0.0.1:5000) and you should be able to view the homepage displaying “Hello world”. Navigate to the registration route `127.0.0.1:5000/register` to view the registration page.
+
+## Install and Setup SQLAlchemy
+To work with SQLAlchemy, we need to install the SQLAlchemy library first.
+
+```bash
+$ pip install SQLAlchemy
+```
+With SQLAlchemy installed, to work with it on our application, we have to import the SQLAlchemy class from the flask_sqlalchemy module in the main application file of our project, which is the __init__.py file in our app folder.
+
+```python
+from flask_sqlalchemy import SQLAlchemy
+```
+
+Next, proceed to configure the variable to the database file. In this case, we are using the SQL database, therefore we make use of SQLite to run queries through our database.
+
+```python
+app.config["SQLALCHEMY_DATABASE_URI"] = 'sqlite:///registration.db'
+``` 
+In the code above, the SQLite uniform resource indicator will receive the name of the database, which in our case is `registration.db`. With this setup, we have to initialize the database connection by creating an object of the SQLAlchemy class, then we pass our application as a parameter.
+
+```python
+db = SQLAlchemy(app)
+``` 
+By following the process above, we have our database initialized and ready to work with, but we can’t add data to this database because it has no tables or columns, neither has it been created for user registration data to be stored. 
+
+However, to solve this problem we have to start working with database models for the application. 
+
+In the process of creating models, we find that some models do have some relationship to one another or with other data, e.g a user might have a specific ID, such that this specific ID also refers back to the user, but how do we tell the database about the existence of this symbiotic relationship? To do this, we first need to understand the basic database relationship and how they are being initialized in our models.
+
+## Database Relationship
+In the process of working with databases, the following are the two types of relationship that you would frequently interact with when working with tables:
+- One to Many
+- One to One
+
+In our understanding of the various database relationships, consider the creation of a Pet management system. The application will consist of the following models: PetHandler, PetCategory, PetComment and the implementation of this model is as follows:
```

Comment:
Before you start explaining the codes, it would be better if you explain the objectives of what you're going to code.